packages/terra-layout/src/_HoverTarget.jsx

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import debounce from 'lodash.debounce';
+import IconChevronRight from 'terra-icon/lib/icon/IconChevronRight';
+import 'terra-base/lib/baseStyles';
+
+import styles from './HoverTarget.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The component to display in the panel content area.
+   */
+  children: PropTypes.node,
+  /**
+   * Enables the hover functionality.
+   */
+  hoverIsEnabled: PropTypes.bool,
+  /**
+   * Disables the hover bar.
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * Function callback to trigger when the hover section is clicked on mobile.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Function callback to trigger when the mouse leaves.
+   */
+  onHoverOff: PropTypes.func,
+  /**
+   * Function callback to trigger when the mouse enters.
+   */
+  onHoverOn: PropTypes.func,
+  /**
+   * String text to display.
+   */
+  text: PropTypes.string,
+};
+
+const defaultProps = {
+  children: [],
+  hoverIsEnabled: false,
+};
+
+class HoverTarget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.setHoverNode = this.setHoverNode.bind(this);
+    this.handleMouseEnter = this.handleMouseEnter.bind(this);
+    this.handleMouseLeave = this.handleMouseLeave.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+
+    this.debouncedUpdateSize = debounce(this.updateSize, 250);
+
+    this.listenersAdded = false;
+    this.isMouseEnterActive = false;
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.debouncedUpdateSize);
+
+    this.updateSize();
+
+    if (this.hoverNode) {
+      this.listenersAdded = this.updateListenersOnNode(this.hoverNode);
+    }
+  }
+
+  componentDidUpdate() {
+    this.updateSize();
+
+    if (this.hoverNode) {
+      this.listenersAdded = this.updateListenersOnNode(this.hoverNode);
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.debouncedUpdateSize);
+
+    if (this.hoverNode) {
+      this.listenersAdded = this.removeListenersFromNode(this.hoverNode);
+    }
+  }
+
+  setHoverNode(node) {
+    this.hoverNode = node;
+  }
+
+  updateSize() {
+    // Due to the nature of the rotated text and therefore inverted coordinates, we need to update
+    // the width of the text element to match the available height of the container.
+    if (this.textElement) {
+      this.textElement.style.width = `${this.textContainerElement.offsetHeight}px`;
```

Comment:
Would there ever be an instance where this.textElement.style.width is undefined and this new width wouldn't set correctly?

Reply:
Shouldn't be, those are all browser-native properties on the node.